File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/04.For-Loop/02-exercise/06.oscars.js
function oscars(input) {

    const pointsForNominee = 1250.5;

    let index = 0;
    let actorName = input[index];
    index++;
    let points = Number(input[index]);
    index++;
    let evaluatorCount = Number(input[index]);
    index++;

    let isNominee = false;

    for (let i = 1; i <= evaluatorCount; i++) {
        let evaluatorName = input[index];
        index++;
        let evaluatorPoints = Number(input[index]);
        index++;
        points += evaluatorName.length * evaluatorPoints / 2;

        if (points > pointsForNominee) {
            console.log(`Congratulations, ${actorName} got a nominee for leading role with ${points.toFixed(1)}!`);
            isNominee = true;
            break;
        }
    }

    const difference = pointsForNominee - points;
    if (!isNominee) {
        console.log(`Sorry, ${actorName} you need ${difference.toFixed(1)} more!`);
    }
}
oscars(["Sandra Bullock", "340", "5", "Robert De Niro", "50", "Julia Roberts", "40.5", "Daniel Day-Lewis", "39.4", "Nicolas Cage", "29.9", "Stoyanka Mutafova", "33"]);

File: Python_Basics/Homeworks-And-Labs/04.for_loop/02.exercise/06.oscars.py
points_for_nominee = 1250.5

actor_name = input()
points = float(input())
evaluator_count = int(input())

is_nominee = False

for i in range(evaluator_count):
    evaluator_name = input()
    evaluator_points = float(input())

    points += len(evaluator_name) * evaluator_points / 2

    if points > points_for_nominee:
        print(f'Congratulations, {actor_name} got a nominee for leading role with {points:.1f}!')
        is_nominee = True
        break

difference = points_for_nominee - points
if not is_nominee:
    print(f'Sorry, {actor_name} you need {difference:.1f} more!')
